src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -502,7 +504,6 @@ class CVulkanLogicalDevice final : public ILogicalDevice
             vk_writeDescriptorSets[i].descriptorType = getVkDescriptorTypeFromDescriptorType(pDescriptorWrites[i].descriptorType);
             vk_writeDescriptorSets[i].descriptorCount = pDescriptorWrites[i].count;
 
-            assert(pDescriptorWrites[i].count <= MAX_DESCRIPTOR_ARRAY_COUNT);
             assert(pDescriptorWrites[i].info[0].desc);
 
             switch (pDescriptorWrites[i].info->desc->getTypeCategory())
```

Comment:
Why on earth are you pushing back dummy structs?

you could have just done `...Infos.resize(...Infos.size()+pDescriptorWrites[i].count);`

(also while probably DRYing a bit and doing `const auto bindingWriteCount  = pDescriptorWrites[i].count;`)

Reply:
This change was in response to this comment https://github.com/Devsh-Graphics-Programming/Nabla/pull/345#discussion_r1047770830

Reply:
yes either approach is correct, what I'm suggesting is just more optimal

But by doing `.resize(.size()+pDescriptorWrites[i].count)` you:
1. have less code
2. avoid the nested `for` loop of `push_back` inside the switch

Reply:
Resolved.